app/models/project_validator.rb

```diff
@@ -1,12 +1,15 @@
 class ProjectValidator < ActiveModel::Validator
     def validate(project)
+        # we need this because this method references the metadata_json which is not updated until the project is saved
+        project.metadata = project.metadata_model
+
         # Required fields, always validate
-        validate_role(project:, netid: project.metadata[:data_manager], role: "Data Manager")
-        validate_role(project:, netid: project.metadata[:data_sponsor], role: "Data Sponsor")
+        validate_role(project:, netid: project.metadata_model.data_manager, role: "Data Manager")
+        validate_role(project:, netid: project.metadata_model.data_sponsor, role: "Data Sponsor")
 
         # Validate if present
-        project.metadata[:data_user_read_only]&.each { |read_only| validate_role(project:, netid: read_only, role: "Data User Read Only")}
-        project.metadata[:data_user_read_write]&.each { |read_write| validate_role(project:, netid: read_write, role: "Data User Read Write")}
+        project.metadata_model.ro_users&.each { |read_only| validate_role(project:, netid: read_only, role: "Data User Read Only")}
```

Comment:
You are changing the name of the database field here.  Was that intended.  If so you need a data migration

Reply:
Good point. We kept getting confused with the names `data_user_read_only` vs `ro_users` (and the read-write versions) so we picked one and ran with it, but later on we noticed that the keys are saved to the database. I didn't think we needed to migrate any data at this point, but we might as well play it save. We'll update the code to use the original keys.